packages/block-library/src/post-template/edit.js

```diff
@@ -203,14 +197,68 @@ export default function PostTemplateEdit( {
 					);
 				}
 			}
+
 			// When we preview Query Loop blocks we should prefer the current
 			// block's postType, which is passed through block context.
 			const usedPostType = previewPostType || postType;
+
+			// Only handle sticky posts for 'post' post type.
+			const shouldHandleSticky = usedPostType === 'post';
+
+			// If sticky is not applicable for this post type, or sticky is not set
+			if ( ! shouldHandleSticky ) {
+				return {
+					posts: getEntityRecords( 'postType', usedPostType, {
+						...query,
+						...restQueryArgs,
+					} ),
+					blocks: getBlocks( clientId ),
+				};
+			}
+
+			// If sticky is not set, it will return all posts in the results.
+			// If sticky is set to `only`, it will limit the results to sticky posts only.
+			// If it is anything else, it will exclude sticky posts from results. For the record the value stored is `exclude`.
+			if ( sticky === 'only' ) {
```

Comment:
Perhaps these two conditions could be combined into one, where the value of sticky depends on if it is only/exclude?
It depends on what everyone prefers in terms of readability, but also pending the introduction of **ignore**.
https://github.com/WordPress/gutenberg/pull/66222, which relies on this issue.